packages/notifier/README.md

```diff
@@ -1,6 +1,263 @@
 # Notifier
 
-Produces a notifier, which allows a service to notify clients of
-state changes using a stream of update promises.
+This package provides two similar abstractions for producing and consuming
+asyncronous sequences of values, the *NotifierKit* and the *SubscriptionKit*.
+Both enable a service to notify clients of state changes.
 
-For details, see the [documentation](https://agoric.com/documentation/distributed-programming.html#notifiers) .
+For user documentation on the NotifierKit, see the
+[documentation](https://agoric.com/documentation/distributed-programming.html#notifiers).
+The following is a more precise description of the semantics of the
+NotifierKit, the SubscriptionKit, and their distributed system properties.
+
+
+# Distributed Asynchronous Iteration
+
+JavaScript defines `AsyncGenerator`s, `AsyncIterable`s, and `AsyncIterator`s.
+Each of these manipulate an *async iteration*, an abstract sequence of
+values with the following structure:
+   * Any number of *non-final values* in a fully ordered sequence to be revealed
+     asynchronously over time. These may continue indefinitely or may be
+     terminated in one of two ways:
+   * The async iteration may *finish* successfully, reporting a final
+     *completion value*, or
+   * The async iteration may *fail*, reporting an alleged *reason* for its
+     failure. The reason is typically an error object, and should be, but may
+     in fact be any JavaScript value.
+
+The call `makeNotifierKit()` makes an `{updater, notifier}` pair.
+The call `makeSubscriptionKit()` for makes a similar
+`{publication, subscription}` pair. The first element of each pair (`updator`,
```

Comment:
updator/updater

Reply:
Done

Reply:
> I think there’s a coherent story for @michaelfig’s idea should we add a solution for that case, commented inline.

See my response to @michaelfig at https://github.com/Agoric/agoric-sdk/blob/async-iter-kit/packages/notifier/test/map-unum.js . Though note the disclaimer at the top.

Reply:
> I don’t suppose we could also have test cases where the producer and consumer are genuinely in separate vats, to give eventual send some inter-vat exercise?

We absolutely need to do that since all our motivating uses cases are inter-vat. Until we do, we cannot be confident that this will serve these intended purposes. It would be good, though much harder, to have instrumented inter-vat tests to verify that these have the distributed systems efficiency properties they are supposed to have. But not in this PR.